cmux.go

```diff
@@ -19,8 +19,12 @@ import (
 	"io"
 	"net"
 	"sync"
+	"time"
 )
 
+// Default timeout for matches
+var DefaultMatchTimeout = time.Minute
```

Comment:
This should be a config parameter not a one-size-fit-all timeout.

Instead, please (a) add a timeout field to `cMux` (the default should be no timeout), (b) add a SetReadTimeout to the `CMux` interface, and (c) add unittests to test the timeout functionality. Thanks.